src/plugins/upf/test/test_upf.py

```diff
@@ -76,17 +85,202 @@ def upf_setup_cmds(cls):
             "upf tdf ul enable ip4 %s" % cls.if_access.name,
             "ip route add 0.0.0.0/0 table 2 via %s %s" %
             (cls.if_sgi.remote_ip4, cls.if_sgi.name),
-            "create upf application name TST",
+            "create upf application proxy name TST",
             "upf application TST rule 3000 add l7 regex " +
             r"^https?://(.*\\.)*(example)\\.com/",
+            # TODO: uncomment when IP rules are implemented
+            # "upf application TST rule 3001 add ipfilter " +
+            # "permit out ip from %s to assigned" % APP_RULE_IP_V4,
         ]
 
+    @property
+    def pfcp_cp_ip(self):
+        return self.if_cp.remote_ip4
+
+    @property
+    def pfcp_up_ip(self):
+        return self.if_cp.local_ip4
+
+    @property
+    def ue_ip(self):
+        return self.if_access.remote_ip4
+
+    @property
+    def sgi_remote_ip(self):
+        return self.if_sgi.remote_ip4
+
+    def ie_ue_ip_address(self, SD=0):
+        return IE_UE_IP_Address(ipv4=self.ue_ip, V4=1, SD=SD)
+
+    def ie_redirect_information(self):
+        return IE_RedirectInformation(
+            type="IPv4 address",
+            address=REDIR_TARGET_IP_V4)
+
+    def ie_sdf_filter_for_redirect(self):
+        return IE_SDF_Filter(
+            FD=1,
+            flow_description="permit out ip from %s to assigned" %
+            REDIR_IP_V4)
+
+    def ie_sdf_filter_extra_for_app_pdi(self):
+        return IE_SDF_Filter(
+            FD=1,
+            flow_description="permit out ip from %s to assigned" %
+            EXTRA_SDF_IP_V4)
+
+    def ie_fseid(self):
+        return IE_FSEID(ipv4=self.pfcp_cp_ip, v4=1, seid=self.cur_seid)
+
+    def verify_ie_fseid(self, ie_fseid):
+        self.assertTrue(ie_fseid.v4)
+        self.assertFalse(ie_fseid.v6)
+        self.assertEqual(ie_fseid.ipv4, self.pfcp_up_ip)
+
+    def verify_ie_ue_ip_address(self, ie_ue_ip):
+        self.assertEqual(ie_ue_ip.V4, 1)
+        self.assertEqual(ie_ue_ip.V6, 0)
+        self.assertEqual(ie_ue_ip.ipv4, self.ue_ip)
+
+    @property
+    def IP(self):
+        return IP
+
+
+class IPv6Mixin(object):
+    drop_ip = DROP_IP_V6
+    redir_ip = REDIR_IP_V6
+    redir_target_ip = REDIR_TARGET_IP_V6
+    non_app_rule_ip = NON_APP_RULE_IP_V6
+    non_app_rule_ip_2 = NON_APP_RULE_IP_2_V6
+    non_app_rule_ip_3 = NON_APP_RULE_IP_3_V6
+    extra_sdf_ip = EXTRA_SDF_IP_V6
+
+    @classmethod
+    def setup_tables(cls):
+        cls.vapi.cli("ip6 table add 1")
+        cls.vapi.cli("ip6 table add 2")
+
+    @classmethod
+    def config_interface(self, test_cls, iface, n):
+        test_cls.vapi.cli("set interface ip6 table %s %d" % (iface.name, n))
+        iface.config_ip6()
+        iface.resolve_ndp()
+
+    @classmethod
+    def upf_setup_cmds(cls):
+        return [
+            "ip6 table add 1",
+            "ip6 table add 2",
```

Comment:
instead of table 1 and 2, could you please use 100 and 200 ?

We had a problem before where the fib indexes and ids where mixed up. With this sequence of command, fib table ids and index will match, hiding potential mixups.